src/test/java/bio/terra/service/snapshotbuilder/SnapshotRequestDaoTest.java

```diff
@@ -92,6 +108,24 @@ void getByIdNotFound() {
     assertThrows(NotFoundException.class, () -> snapshotRequestDao.getById(UUID.randomUUID()));
   }
 
+  @Test
+  void getByCreatedSnapshotId() {
+    SnapshotAccessRequestModel response = createRequest();
+    snapshotRequestDao.updateCreatedInfo(
+        response.id(), createdSnapshot.getId(), "name", "userEmail");
+    SnapshotAccessRequestModel retrieved =
+        snapshotRequestDao.getByCreatedSnapshotId(createdSnapshot.getId());
+    verifyResponseContents(
+        retrieved, List.of("createdSnapshotId", "samGroupName", "samGroupCreatedBy"));
+  }
+
+  @Test
+  void getByCreatedSnapshotIdNotFound() {
+    assertThrows(
+        NotFoundException.class,
+        () -> snapshotRequestDao.getByCreatedSnapshotId(UUID.randomUUID()));
```

Comment:
Can you fix all the sonar warnings in the new code before merging? In this case, sonar will complain that the `assertThrows()` function contains more than one function call. To fix it, you have to move `UUID.randomUUID()` out of the lambda:
```suggestion
    UUID id = UUID.randomUUID();
    assertThrows(
        NotFoundException.class, () -> snapshotRequestDao.getByCreatedSnapshotId(id));
```

There are many test cases where you generate a UUID; you could generate it once as a static field instead, to avoid having to declare the `id` in each `assertThrows()` case.